crates/oxc_linter/src/rules/react/no_unknown_property.rs

```diff
@@ -0,0 +1,736 @@
+        ("shape-rendering", "shapeRendering"),
+        ("stop-color", "stopColor"),
+        ("stop-opacity", "stopOpacity"),
+        ("strikethrough-position", "strikethroughPosition"),
+        ("strikethrough-thickness", "strikethroughThickness"),
+        ("stroke-dasharray", "strokeDasharray"),
+        ("stroke-dashoffset", "strokeDashoffset"),
+        ("stroke-linecap", "strokeLinecap"),
+        ("stroke-linejoin", "strokeLinejoin"),
+        ("stroke-miterlimit", "strokeMiterlimit"),
+        ("stroke-opacity", "strokeOpacity"),
+        ("stroke-width", "strokeWidth"),
+        ("text-anchor", "textAnchor"),
+        ("text-decoration", "textDecoration"),
+        ("text-rendering", "textRendering"),
+        ("underline-position", "underlinePosition"),
+        ("underline-thickness", "underlineThickness"),
+        ("unicode-bidi", "unicodeBidi"),
+        ("unicode-range", "unicodeRange"),
+        ("units-per-em", "unitsPerEm"),
+        ("v-alphabetic", "vAlphabetic"),
+        ("v-hanging", "vHanging"),
+        ("v-ideographic", "vIdeographic"),
+        ("v-mathematical", "vMathematical"),
+        ("vector-effect", "vectorEffect"),
+        ("vert-adv-y", "vertAdvY"),
+        ("vert-origin-x", "vertOriginX"),
+        ("vert-origin-y", "vertOriginY"),
+        ("word-spacing", "wordSpacing"),
+        ("writing-mode", "writingMode"),
+        ("x-height", "xHeight"),
+        ("xlink:actuate", "xlinkActuate"),
+        ("xlink:arcrole", "xlinkArcrole"),
+        ("xlink:href", "xlinkHref"),
+        ("xlink:role", "xlinkRole"),
+        ("xlink:show", "xlinkShow"),
+        ("xlink:title", "xlinkTitle"),
+        ("xlink:type", "xlinkType"),
+        ("xml:base", "xmlBase"),
+        ("xml:lang", "xmlLang"),
+        ("xml:space", "xmlSpace"),
+    ]);
+}
+
+///
+/// Checks if an attribute name is a valid `data-*` attribute:
+/// if the name starts with "data-" and has alphanumeric words (browsers require lowercase, but React and TS lowercase them),
+/// not start with any casing of "xml", and separated by hyphens (-) (which is also called "kebab case" or "dash case"),
+/// then the attribute is a valid data attribute.
+///
+fn is_valid_data_attr(name: &str) -> bool {
+    static DATA_ATTR_REGEX: Lazy<Regex> = Lazy::new(|| Regex::new(r"^data(-?[^:]*)$").unwrap());
+
+    !name.to_lowercase().starts_with("data-xml") && DATA_ATTR_REGEX.is_match(name)
+}
+
+fn normalize_attribute_case(name: &str) -> &str {
+    DOM_PROPERTIES_IGNORE_CASE
+        .iter()
+        .find(|camel_name| camel_name.to_lowercase() == name.to_lowercase())
+        .unwrap_or(&name)
+}
+fn has_uppercase(name: &str) -> bool {
+    name.contains(char::is_uppercase)
+}
+
+impl Rule for NoUnknownProperty {
+    fn from_configuration(value: serde_json::Value) -> Self {
+        let value = value.as_array().and_then(|arr| arr.first());
+
+        Self(Box::new(serde_json::from_value(value.unwrap().clone()).unwrap()))
```

Comment:
is it possilble to change these to `and_then` or `unwrap_or_default` if there's an invalid config, we don't really want to panic 🙂

Reply:
got it. will fix it